test/fw/ptl/lib/pbs_testlib.py

```diff
@@ -9271,14 +9262,12 @@ def create_vnodes(self, name=None, attrib=None, num=1, mom=None,
         # only expect if vnodes were added rather than the nat vnode modified
         if expect and num > 0:
             attrs = {}
-            n = num + len(existing_vnodes)
-            if len(existing_vnodes) > 0:
-                n = n - 1
-            for k, v in attrib.items():
-                attrs[str(k) + '=' + str(self.utils.decode_value(v))] = n
-            attrs['state=free'] = n
-            attrs['resources_available.host=' + natvnode] = n
-            rv = self.expect(VNODE, attrs, attrop=PTL_AND)
+            attrs.update({'state': 'free'})
```

Comment:
If you are only setting one value, why use update() instead of directly setting the value?